libmscore/instrchange.h

```diff
@@ -42,9 +44,17 @@ class InstrumentChange final : public TextBase  {
       void setInstrument(Instrument&& i)    { *_instrument = i;    }
       void setInstrument(const Instrument& i);
 
+      std::vector<KeySig*> keySigs() const  { return _keySigs;     }
```

Comment:
What's the purpose of these arrays (`_keySigs` and `_clefs`)? What are you planning on doing with the elements referenced in them?

Reply:
This is so that if an instrument change is deleted, the key changes and clefs that were created with them can be deleted too. Previous to doing this, there was an issue where if you deleted an instrument change for a transposing instrument, the key change would automatically be hidden, as it wasn't changing the key any more, but it would still be there, and if you added another key change previous to this it would reappear. It has to be a vector as if you change instrument for a part that has more than one line, it inserts a key change for each line. While admittedly this is not a common occurrence, I think its something that should be accounted for.

Reply:
I have to say, I'm not a massive fan of the idea of storing pointers to elements that may be deleted by other parts of the code... however, I can see that it isn't easy to do it any other way. I was going to suggest looping over all 'generated' keysigs after the instrument change, but I can see that that would create more problems than it would solve (i.e. how do you know which ones are created by an instrument change and which are keysigs generated for other reasons? etc.)

So, I think that's OK, then. Just checking, are the new keysigs marked as 'generated' when they're created by adding an instrument change? IIRC that flag ought to be set for them...

Reply:
I don't think they are. I'll change that, and test to make sure there aren't any unwanted side effects.